erpnext/stock/stock_ledger.py

```diff
@@ -589,6 +589,15 @@ def process_sle(self, sle):
 		sle.stock_queue = json.dumps(self.wh_data.stock_queue)
 		sle.stock_value_difference = stock_value_difference
 		sle.doctype = "Stock Ledger Entry"
+
+		if (
+			sle.voucher_type in ["Purchase Receipt", "Purchase Invoice"]
+			and sle.voucher_detail_no
+			and sle.actual_qty < 0
+			and frappe.get_cached_value(sle.voucher_type, sle.voucher_no, "is_internal_supplier")
+		):
+			sle.outgoing_rate = get_incoming_rate_for_inter_company_transfer(sle)
```

Comment:
The `outgoing_rate` should be set at the beginning of the `process_sle` function, because otherwise the system will not consider this outgoing_rate for calculating `valuation_rate`